backend/src/routes/index.ts

```diff
@@ -72,30 +75,44 @@ const routes: FastifyPluginAsync = async (server) => {
         let customPaymasters = [];
         let privateKey = '';
         let supportedNetworks;
+        let noOfTxns;
+        let txnMode;
+        let indexerEndpoint;
         if (!unsafeMode) {
           const AWSresponse = await client.send(
             new GetSecretValueCommand({
               SecretId: prefixSecretId + api_key,
             })
           );
           const secrets = JSON.parse(AWSresponse.SecretString ?? '{}');
-          if (!secrets['PRIVATE_KEY']) return reply.code(ReturnCode.FAILURE).send({ error: ErrorMessage.INVALID_API_KEY })
+          if (!secrets['PRIVATE_KEY']) {
+            server.log.info("Invalid Api Key provided")
+            return reply.code(ReturnCode.FAILURE).send({ error: ErrorMessage.INVALID_API_KEY })
+          }
           if (secrets['ERC20_PAYMASTERS']) {
             const buffer = Buffer.from(secrets['ERC20_PAYMASTERS'], 'base64');
             customPaymasters = JSON.parse(buffer.toString());
           }
           privateKey = secrets['PRIVATE_KEY'];
           supportedNetworks = secrets['SUPPORTED_NETWORKS'];
+          noOfTxns = secrets['NO_OF_TRANSACTIONS_IN_A_MONTH'] ?? 10;
+          txnMode = secrets['TRANSACTION_LIMIT'] ?? 0;
+          indexerEndpoint = secrets['INDEXER_ENDPOINT'] ?? "http://localhost:3003";
         } else {
           const record: any = await getSQLdata(api_key);
```

Comment:
What happens when this throws an error?

Reply:
we have an try/catch block inside getSQLdata fn and as well as in this request fn so if something goes wrong on fetching the data from sql, it ignores this mode and the transaction would go through

Reply:
So i had a look here: https://github.com/etherspot/arka/blob/75bc1a07026a88e3101810aae410437aedbf2af6/backend/src/routes/index.ts#L404

It will throw an error but i can't see where the check is for the error state in this file?